### PR TITLE
Feature/unr 4260 add worker info to boundary 4.24 compatible

### PR DESCRIPTION
**Pull request by Vix2021**
#### Description
The task was to add worker names and virtual worker ids to the worker boundaries. This was achieved by rendering a dynamic material that combines the background colour and the foreground worker information and repeat the material across the worker boundary faces.
[Copied changes from 4.25 to 4.24 for backward compatibility]

Handover Gym:
DynamicMaterialFinal
Teleport Gym:
DynamicMaterialFinal2

Blueprint changes:
TranslucentWorkerRegion (minor change): Change to unlit material to avoid gradient which is visible when tiling the material
translucentworkerregionBP
WorkerRegionCombinedMaterial (new material):
Overview:
image
Part 1:
workerregioncombinedmaterialBP-p1
Part 2:
workerregioncombinedmaterialBP-p2
MuliFont (new Font)
mulifont

#### Release note
- You can now see worker information displayed on the worker's boundaries. The worker name and virtual worker id displayed corresponds to the worker that you are currently looking at and will be visible when you are near a border.

#### Tests
Tested in the GDK test gyms: handover and teleport with different load balancing strategies.
Tested in NWX to check visibility with the different day and night cycles.

STRONGLY SUGGESTED: How can this be verified by QA?
Needs corresponding GDK Test Gym PR to verify the changes: https://github.com/spatialos/UnrealGDKTestGyms/pull/213
Run a zoned map and check that the boundary information is visible and correct for each worker.

#### Documentation
Updated images/animations of the worker boundaries in the multi worker document: https://docs.google.com/document/d/1G9cRz891Ycaiax-M7Psx3rlfX8qTIi0CTk8_71PL4Zg/edit#heading=h.j4ppr52lw6ck

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkerRegion.cpp [hunk 1]
```diff
@@ -24,16 +27,54 @@ AWorkerRegion::AWorkerRegion(const FObjectInitializer& ObjectInitializer)
...
+	}
+
 	SetOpacity(Opacity);
 	SetColor(Color);
 	SetPositionAndScale(Extents, VerticalScale);
+
+	if (!bInEditor)
+	{
+		// At runtime, calls DrawToCanvasRenderTarget to render the dynamic boundary material
+		CanvasRenderTarget->UpdateResource();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkerRegion.cpp:62]: How is this texture updated when running in editor?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp [hunk 2]
```diff
@@ -198,8 +220,10 @@ void ASpatialDebugger::CreateWorkerRegions()
 	for (const FWorkerRegionInfo& WorkerRegionData : WorkerRegions)
 	{
 		AWorkerRegion* WorkerRegion = GetWorld()->SpawnActor<AWorkerRegion>(SpawnParams);
-		WorkerRegion->Init(WorkerRegionMaterial, WorkerRegionData.Color, WorkerRegionOpacity, WorkerRegionData.Extents, WorkerRegionHeight,
-						   WorkerRegionVerticalScale);
+		FString WorkerInfo = FString::Printf(TEXT("You are looking at virtual worker number %d\n%s"), WorkerRegionData.VirtualWorkerID,
+											 *WorkerRegionData.WorkerName);
+		WorkerRegion->Init(WorkerRegionMaterial, WorkerCombinedMaterial, WorkerInfoFont, WorkerRegionData.Color, WorkerRegionOpacity,
+						   WorkerRegionData.Extents, WorkerRegionHeight, WorkerRegionVerticalScale, WorkerInfo, bInEditor);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:226]: Instead of passing `bInEditor` around, couldn't we just deduce it?

**Review by m-samiec — APPROVED** [review]
Couple of comments - looks great!

File: SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkerRegion.cpp (hunk 1 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkerRegion.cpp:62]: This approach to dynamically render the text on the material via a Canvas only works at runtime as the UCanvasRenderTarget2D->OnCanvasRenderTargetUpdate() callback is only triggered at runtime.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp (hunk 2 above)
Comment (Vix2021) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp:226]: I haven't found a way to deduce if the class is created in the editor or at runtime. However, as the UCanvasRenderTarget2D->OnCanvasRenderTargetUpdate() callback is only triggered at runtime this bool can be removed.

**Pull request merged**